### PR TITLE
Feature/signature

### PR DESCRIPTION
**Pull request by denaskhabovG**

### CODE REVIEW

File: src/app.controller.ts [hunk 1]
```diff
@@ -7,6 +7,6 @@ export class AppController {
 
   @Get()
   getHello(): string {
-    return this.appService.getHello();
+    return 'asasas';
```
Comment (IslamRustamov1) [inline src/app.controller.ts]: remove this controller completely

File: src/app.service.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import { Injectable } from '@nestjs/common';
+import { Sequelize } from 'sequelize-typescript';
 
 @Injectable()
 export class AppService {
```
Comment (IslamRustamov1) [inline src/app.service.ts]: remove this service completely

File: src/config/SequelizeConfigService.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { UserSignature } from '../userSignature/models/userSignature.model';
+
+@Injectable()
+export class SequelizeConfigService implements SequelizeOptionsFactory {
+  constructor(private readonly configService: ConfigService) {}
+
+  createSequelizeOptions(): SequelizeModuleOptions {
+    const {
+      pg: { dialect, host, port, username, password, database, logging },
+    } = this.configService.get<IDatabaseConfig>('database');
```
Comment (IslamRustamov1) [inline src/config/SequelizeConfigService.ts]: ('database'); move this to enum

File: src/config/configuration.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    host: string;
+    port: number;
+    username: string;
+    password: string;
+    database: string;
+    logging: boolean;
+  };
+}
+
+export const databaseConfig = registerAs('database', (): IDatabaseConfig => ({
```
Comment (IslamRustamov1) [inline src/config/configuration.ts]: take 'database' string from enum

File: src/session/session.module.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { Module } from '@nestjs/common';
+import { SequelizeModule } from '@nestjs/sequelize';
+import { Session } from './models/session.model';
+import { ConfigModule } from '@nestjs/config';
```
Comment (IslamRustamov1) [inline src/session/session.module.ts:3]: Import sequence - first comes official libraries, then yours

File: src/todos/dto/change-todo.dto.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export class ChangeTodoDto {
+  readonly title: string;
+  readonly done: boolean;
```
Comment (IslamRustamov1) [inline src/todos/dto/change-todo.dto.ts:9]: Add decorators for type validation

File: src/todos/dto/create-todo.dto.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export class CreateTodoDto {
+  readonly title: string;
+  readonly done: boolean;
```
Comment (IslamRustamov1) [inline src/todos/dto/create-todo.dto.ts:9]: Add decorators for type validation

File: src/todos/dto/update-todo.dto.ts [hunk 8]
```diff
@@ -0,0 +1,4 @@
+export class UpdateTodoDto {
+  readonly title: string;
+  readonly done: boolean;
```
Comment (IslamRustamov1) [inline src/todos/dto/update-todo.dto.ts]: Add decorators for type validation

File: src/todos/dto/create-todo.dto.ts (hunk 7 above)
Comment (IslamRustamov1) [inline src/todos/dto/create-todo.dto.ts:9]: remove done from create todo dto

File: src/todos/todos.controller.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  @Get(':id')
+  getOneTodo(@Param('id') id: string) {
+    return this.todoService.findOne(id);
+  }
+
+  @Post()
+  @HttpCode(HttpStatus.CREATED)
+  @Header('Content-Type', 'application/json')
```
Comment (IslamRustamov1) [inline src/todos/todos.controller.ts:22]: check if this is truly needed

File: src/todos/todos.controller.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    return this.todoService.create(createTodo);
+  }
+
+  @Delete(':id')
+  deleteTodo(@Param('id') id: string) {
+    return this.todoService.remove(id);
+  }
+
+  @Patch()
+  changeTodo(@Body() updateTodo: UpdateTodoDto, @Param('id') id: string) {
```
Comment (IslamRustamov1) [inline src/todos/todos.controller.ts]: Id must be passed here

File: src/userSignature/dto/create-signature.dto.ts [hunk 11]
```diff
@@ -0,0 +1,3 @@
+export class SignInDto {
+  readonly text: string;
```
Comment (IslamRustamov1) [inline src/userSignature/dto/create-signature.dto.ts:5]: need decorators

File: src/userSignature/userSignature.controller.ts [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    return this.userSignatureService.getSignature();
+  }
+
+  @Post()
+  signIn(@Body() signature: SignInDto) {
+    return this.userSignatureService.signIn(signature);
+  }
+
+  @Delete(':id')
+  deleteTodo(@Param('id') id: number) {
```
Comment (IslamRustamov1) [inline src/userSignature/userSignature.controller.ts]: I don't think that this fits here

File: src/userSignature/userSignature.service.ts [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+    private UserSignatureModel: typeof UserSignature,
+  ) {
+  }
+
+  encrypt(message = '', key = '') {
+    const msg = CryptoJS.AES.encrypt(message, key);
+    return msg.toString();
+  }
+
+  decrypt(message = '', key = '') {
```
Comment (IslamRustamov1) [inline src/userSignature/userSignature.service.ts]: move encrypt and decrypt to a different service

File: src/userSignature/userSignature.service.ts [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+    try {
+      const signature1 = body.text;
+      const signature2 = await this.UserSignatureModel.findOne({
+        where: {
+          text: body.text,
+        },
+      });
+
+      const decryptedSignature1 = this.decrypt(signature1, 'secret');
+      const decryptedLastSignature = this.decrypt(signature2.text, 'secret');
```
Comment (IslamRustamov1) [inline src/userSignature/userSignature.service.ts]: rename this to decryptedSignature2

**Review by IslamRustamov1 — CHANGES_REQUESTED** [review]

File: src/userSignature/userSignature.controller.ts (hunk 12 above)
Comment (denaskhabovG) [reply src/userSignature/userSignature.controller.ts:20]: Do i need delete todo by id?

**Review by IslamRustamov1 — APPROVED** [review]

**Pull request merged**